src/configuration.js

```diff
@@ -1,12 +1,29 @@
+const cosmiconfig = require('cosmiconfig');
+import { readSync, readFileSync } from 'fs';
+import getGraphQLProjectConfig from 'graphql-config';
+import path from 'path';
+
 import defaultRules from './rules/index.js';
 import JSONFormatter from './formatters/json_formatter.js';
 import TextFormatter from './formatters/text_formatter.js';
-import getGraphQLProjectConfig from 'graphql-config';
-import { readSync, readFileSync } from 'fs';
 
 export class Configuration {
-  constructor(options, stdinFd) {
-    this.options = options;
+  /*
+    options:
+      - configDirectory: path to begin searching for config files
+      - except: [string array] blacklist rules
+      - format: (required) `text` | `json`
+      - only: [string array] whitelist rules
+      - schemaFileName: [string] file to read schema from
+      - stdin: [boolean] pass schema via stdin?
+  */
+  constructor(options = {}, stdinFd = null) {
+    const defaultOptions = { format: 'text' };
+    const configOptions = loadOptionsFromConfig(
+      options.configDirectory || process.cwd()
```

Comment:
Rather than use `process.cwd()` do you think we should make this class completely agonistic to the env?

i.e.

In `src/cli.js` we'd do:

```diff
- const exitCode = run(process.stdout, process.stdin, process.argv);
+ const exitCode = run(process.stdout, process.stdin, process.argv, process.cwd());
```

and `src/configuration.js`:

```diff
- constructor(options = {}, stdinFd = null) {
+ constructor(options = {}, stdinFd = null, currentWorkingDir) {
```

My goal was to keep all the env-related things in `cli.js` that way all the other classes can be easily tested and do not rely on `process`. Thoughts?